### PR TITLE
METRON-1436: Manually Install Solr Cloud in Full Dev

### PR DESCRIPTION
**Pull request by mmiklavc**
https://issues.apache.org/jira/browse/METRON-1436

Provide script for starting Solr Cloud in quickstart mode for testing purposes

## Contributor Comments
Adds a solr installation script to the solr RPM. New Solr README includes documentation for manually installing the RPM and running the script to spin up Solr Cloud. Expect follow-on work to get a more production-ready deployment version (e.g. with system service scripts). Preliminary notes are provided in the README and installation script.

## Pull Request Checklist

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-platform/metron-solr/src/main/scripts/install_solr.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# create user if not exists
+solr_uid="`id -u "$SOLR_USER"`"
+if [ $? -ne 0 ]; then
+  echo "Creating new user: $SOLR_USER"
+  adduser --system -U -m --home-dir "$SOLR_VAR_DIR" "$SOLR_USER"
+fi
+cd $SOLR_VAR_DIR
+wget http://archive.apache.org/dist/lucene/solr/${SOLR_VERSION}/solr-${SOLR_VERSION}.tgz
+tar zxvf solr-${SOLR_VERSION}.tgz
+chown -R solr:solr solr-${SOLR_VERSION}
```
Comment (JonZeolla) [inline metron-platform/metron-solr/src/main/scripts/install_solr.sh]: `chown -R $SOLR_USER:$SOLR_USER solr-${SOLR_VERSION}`

File: metron-platform/metron-solr/src/main/scripts/install_solr.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+cd solr-${SOLR_VERSION}
+su solr -c "bin/solr -e cloud -noprompt"
```
Comment (JonZeolla) [inline metron-platform/metron-solr/src/main/scripts/install_solr.sh]: `su $SOLR_USER -c "bin/solr -e cloud -noprompt"`

File: metron-platform/metron-solr/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# Solr in Metron
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Installing](#installing)
+
+## Introduction
+
+Metron ships with Solr 6.6.2. Solr Cloud can be used as the real-time portion of the datastore resulting from [metron-indexing](../metron-indexing/README.md).
```
Comment (justinleet) [inline metron-platform/metron-solr/README.md]: Can you make this "... Solr 6.6.2 support", so it's obvious we're not actually including Solr?

I'd also add the blurb / link from the script about production use here as well.

**Comment by merrimanr** [comment]
I tested it out and worked fine.  I think it's a good start.  +1 as long as other commenters are satisfied.

File: metron-platform/metron-solr/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+```
+cd /localrepo
+ls -al
+```
+
+Copy the name of the Solr RPM and install it via the following command.
+
+```
+# Install Solr RPM - replace the Metron version and appropriate timestamp on the rpm
+rpm -ivh metron-solr-0.4.3-201801301626.noarch.rpm
```
Comment (justinleet) [inline metron-platform/metron-solr/README.md]: Could this just be the following?
```
cd /localrepo
rpm -ivh metron-solr*.rpm
```

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec [hunk 5]
```diff
@@ -225,6 +225,7 @@ This package installs the Metron Solr files
 %dir %{metron_home}/bin
 %dir %{metron_home}/config
 %dir %{metron_home}/lib
+%{metron_home}/bin/install_solr.sh
```
Comment (justinleet) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: There should also be an entry at the bottom for the changelog.

**Comment by justinleet** [comment]
I got full dev spinning spun up with this, and it looks good.  I think the latest couple comments are still to be addressed, but otherwise good to go.

**Comment by justinleet** [comment]
Thanks for the updates.  I'm +1 on including this in the feature branch.

**Pull request closed without merging**